### ISSUE TITLE
Locale de_xy: house numbers are after street name in Europe, not before

### ISSUE THREAD

**Issue by Slooti** [faker-js/faker#1524]
### Pre-Checks

- [X] Follow our [Code of Conduct](https://github.com/faker-js/faker/blob/next/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/faker-js/faker/blob/next/CONTRIBUTING.md).
- [X] Read the [docs](https://fakerjs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/faker-js/faker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Faker issue and not related to a combination with another package.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/faker-js/faker/discussions) or join our [Discord Chat Server](https://chat.fakerjs.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
- [ ] I am willing to provide a PR.

### Describe the bug

**Issue**
When generating addresses in Europe (set locale to 'de' or 'de_CH', street addresses should be in the format 'streetName houseNumber'.
Correct: Deutlerweg 12
Wrong: 12 Deutlerweg

**Steps to reproduce**
```
faker.setLocale('de_CH')
const address = faker.address.streetAddress()
```

**Current behaviour**
 "address": "09336 Reinhard Trail",

**Expect behaviour**
 "address": "Reinhard Trail 9336"

### Minimal reproduction code

```
faker.setLocale('de_CH')
const address = faker.address.streetAddress()
```

### Additional Context
To be super correct, the format actually should be in the format of the country that the address is generated in. However this is not possible as generation is not aware of the country at the time of address generation. It makes sense to implement the change, as if I am a Swiss developer, I most likely target the Swiss market and will generate fake data targeted to Switzerland.

### Environment Info

```shell
System:
    OS: macOS 13.0
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 35.21 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.10.0 - ~/node_modules/.bin/npm
  Browsers:
    Brave Browser: 107.1.45.118
    Chrome: 107.0.5304.87
    Firefox: 106.0.3
    Safari: 16.1
```

### Which module system do you use?

- [ ] CJS
- [X] ESM

### Used Package Manager

npm

**Comment by ST-DDT** [comment]
The issue here is that de_CH does not have an address.street_address definition.
You can avoid/fix that by adding the missing entry to the definitions or setting the fallback locale to `de`.

````ts
faker.locales.de_CH.address.street_address =
  faker.locales.de.address.street_address;

// or
faker.localeFallback = "de";
````

In v8 we change how the default locales are provided/build so the workaround will no longer be needed as the de_CH locale will use the de locale as fallback by default (followed by the en fallback as right now)

**Comment by ST-DDT** [comment]
Will be fixed by #1735

**Comment by Slooti** [comment]
Thanks a lot!!

cheers

Martin
